cannon/mipsevm/tests/helpers.go

```diff
@@ -50,31 +50,73 @@ func multiThreadElfVmFactory(t require.TestingT, elfFile string, po mipsevm.Prei
 	return fpvm
 }
 
+type ProofGenerator func(t require.TestingT, state mipsevm.FPVMState, memoryProofAddresses ...uint32) []byte
+
+func singalThreadedProofGenerator(t require.TestingT, state mipsevm.FPVMState, memoryProofAddresses ...uint32) []byte {
+	stState, ok := state.(*singlethreaded.State)
+	if !ok {
+		require.Fail(t, "Failed to cast FPVMState to singlethreaded State type")
+	}
+
```

Comment:
No need to cast the state here since we're not using any methods specific to the singlethreaded implementation. 
```suggestion
```